lib/runtime/text-encoder.js

```diff
@@ -0,0 +1,25 @@
+const textEncoding = require('text-encoding')
+
+function CustomTextDecoder (label, options) {
+  if (!(this instanceof CustomTextDecoder)) {
+    throw TypeError('Called as a function. Did you forget \'new\'?')
+  }
+  this.decoder = new textEncoding.TextDecoder(label, options)
+  return this
+}
+
+CustomTextDecoder.prototype.decode = function decode (input, options) {
```

Comment:
We're missing a few properties on CustomTextDecoder that are present in the [Encoding Standard ](https://encoding.spec.whatwg.org/#textdecodercommon)and on `text-encoding`'s [TextDecoder implementation](https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L984)

Reply:
Hi there,

Sorry I'm not following what you mean.  Apologies.   The `label` and `options` are being passed through to the upstream `textEncoding.TextDocoder` on line 7 (`this.decoder = new textEncoding.TextDecoder(label, options)`).

Which other properties are you referring to? (apologies if I'm not following here)

Reply:
No worries. In the standard, they mention the following properties:
```
  readonly attribute DOMString encoding;
  readonly attribute boolean fatal;
  readonly attribute boolean ignoreBOM;
```

In `text-encoding`'s implementation, you can seem them defined starting here:
https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L1051

Reply:
do they not get set via the options passed into the constructor?

- https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L1026
- https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L1030
- https://github.com/inexorabletash/text-encoding/blob/master/lib/encoding.js#L1035

Reply:
They do but `CustomDecoder` doesn't pass gets to those properties to the underlying `TextDecoder`. For example, invoking `ignoreBOM` on an instance of CustomDecoder returns `undefined`. Instead, it should return `this.decoder.ignoreBOM`

Reply:
Ah, righto.  Now I think I understand.
Let me have a look.

Thanks for explaining.
Cheers
/dom